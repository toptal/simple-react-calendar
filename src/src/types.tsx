import * as Luxon from 'luxon'

export type CalendarSelectionI = CalendarSelectionRangeI | CalendarSelectionSingleI

export interface CalendarRangeSelectedDate extends CalendarSelectionRangeI {
  inProgress: boolean
}

export interface CalendarSelectionRangeI {
  start: Luxon.DateTime
  end: Luxon.DateTime
}

export type CalendarSelectionSingleI = Luxon.DateTime

export enum CalendarSelectionTypeEnum {
  start = 'start',
  end = 'end',
}

export enum CalendarTypeEnum {
  single = 'single',
  range = 'range',
}

export enum MonthHeaderButtonTypeEnum {
  next = 'next',
  prev = 'previous',
}

export enum NoticeTypeEnum {
  overlap = 'overlapping_with_disabled',
  disabled = 'disabled_day_click',
}
