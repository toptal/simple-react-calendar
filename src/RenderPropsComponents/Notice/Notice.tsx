import React, { FC } from 'react'

type Props = {
  blockClassName?: string
}

const displayName = 'Notice'

const Notice: FC<Props> = ({ blockClassName, children }) => {
  if (!children) return null

  return <div className={`${blockClassName}-notice`}>{children}</div>
}

Notice.displayName = displayName

export default Notice
