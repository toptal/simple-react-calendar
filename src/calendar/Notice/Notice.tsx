import React, { FC } from 'react'

import { INoticeType } from '../../@types'

export interface Props {
  blockClassName: string
  type: INoticeType
}

const Notice: FC<Props> = ({type, blockClassName}) => {
  const content =
    type === 'overlapping_with_disabled' ||
    (type === 'disabled_day_click' && 'Selected range must not overlap with disabled dates.')

  return <div className={`${blockClassName}-notice`}>{content}</div>
}

Notice.displayName = 'Notice'

export default Notice
