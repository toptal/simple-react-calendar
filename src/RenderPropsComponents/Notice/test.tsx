import { render } from '@testing-library/react'
import React, { ComponentProps } from 'react'

import Notice from './Notice'

const renderNotice = (props: Partial<ComponentProps<typeof Notice>> = {}) =>
  render(<Notice {...props} />)

test('default render', () => {
  const { container } = renderNotice({ children: 'Example notice error', blockClassName: 'example-class-name' })

  expect(container).toMatchSnapshot()
})

describe('when no `children` defined', () => {
  test('renders null', () => {
    const { container } = renderNotice({ })

    expect(container).toMatchSnapshot()
  })
})
