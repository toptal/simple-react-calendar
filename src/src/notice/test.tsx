import {shallow} from 'enzyme'
import * as React from 'react'

import Notice, {SimpleReactCalendarNoticeProps} from '.'
import {NoticeTypeEnum} from '../types'

describe('<Notice />', () => {
  let props, wrapper

  context('when prop `type` is "overlap"', () => {
    it('renders <Notice />', () => {
      props = getProps()
      wrapper = shallow(<Notice {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  context('when prop `type` is "disabled"', () => {
    it('renders <Notice />', () => {
      props = getProps({type: NoticeTypeEnum.disabled})
      wrapper = shallow(<Notice {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})

const getProps = (overrides: Partial<SimpleReactCalendarNoticeProps> = {}): SimpleReactCalendarNoticeProps => ({
  blockClassName: 'example-class-name',
  type: NoticeTypeEnum.overlap,
  ...overrides,
})
