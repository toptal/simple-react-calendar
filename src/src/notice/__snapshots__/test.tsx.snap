// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notice /> when prop \`type\` is "disabled" renders <Notice /> 1`] = `
<div
  className="example-class-name-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`;

exports[`<Notice /> when prop \`type\` is "overlap" renders <Notice /> 1`] = `
<div
  className="example-class-name-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`;
